lib/User/UserEntry.php

```diff
@@ -0,0 +1,360 @@
+	 * bytes), '1234 MB' (quota in MB - check the \OC_Helper::computerFileSize method for more info)
+	 *
+	 * @return string quota
+	 */
+	public function getQuota() {
+		$quota = null;
+
+		$attr = $this->getAttributeName('ldapQuotaAttribute');
+		if ($attr === '') {
+			\OC::$server->getLogger()->debug("No LDAP quota attribute configured", ['app' => 'user_ldap']);
+		} else {
+			$quota = $this->getAttributeValue ($attr);
+			if (!$this->verifyQuotaValue($quota)) {
+				\OC::$server->getLogger()->debug("Invalid quota <$quota> for LDAP user <{$this->getOwnCloudUID()}>", ['app' => 'user_ldap']);
+				$quota = null;
+			}
+		}
+
+		if ($quota === null) {
+			if (!$this->connection->ldapQuotaDefault) {
+				\OC::$server->getLogger()->debug("No LDAP quota default configured", ['app' => 'user_ldap']);
+			} else {
+				$quota = $this->connection->ldapQuotaDefault;
+				if (!$this->verifyQuotaValue($quota)) {
+					\OC::$server->getLogger()->debug("Invalid default quota <$quota>", ['app' => 'user_ldap']);
+					$quota = null;
+				}
+			}
+		}
+		if ($quota === null) {
+			$quota = 'default';
+		}
+
+		\OC::$server->getLogger()->debug("using quota <$quota> for user <{$this->getOwnCloudUID()}>", ['app' => 'user_ldap']);
+		return $quota;
+
+	}
+
+	private function verifyQuotaValue($quotaValue) {
+		return $quotaValue === 'none' || $quotaValue === 'default' || \OC_Helper::computerFileSize($quotaValue) !== false;
+	}
+
+	/**
+	 * @return string|null email
+	 */
+	public function getEMailAddress() {
+		$attr = $this->getAttributeName('ldapEmailAttribute', 'mail');
+		return $this->getAttributeValue ($attr);
+	}
+	/**
+	 * returns the home directory of the user if specified by LDAP settings
+	 * @return string|null
+	 * @throws \Exception
+	 */
+	public function getHome() {
+		$path = '';
+ 		$attr = $this->getAttributeName('homeFolderNamingRule', null);
+		if (is_string($attr) && strpos($attr, 'attr:') === 0 // TODO do faster startswith check
+			&& strlen($attr) > 5
+		) {
+			$attr = substr($attr, 5);
+
+			$path = $this->getAttributeValue($attr, '');
+		}
+
+		if ($path !== '') {
+			//if attribute's value is an absolute path take this, otherwise append it to data dir
+			//check for / at the beginning
+			if('/' !== $path[0]) {
+				$path = $this->config->getSystemValue('datadirectory',
+						\OC::$SERVERROOT.'/data' ) . '/' . $path;
+			}
+			return $path;
+		}
+
+		if(!is_null($attr)
+			&& $this->config->getAppValue('user_ldap', 'enforce_home_folder_naming_rule', true)
+		) {
+			// a naming rule attribute is defined, but it doesn't exist for that LDAP user
+			throw new \Exception('Home dir attribute can\'t be read from LDAP for uid: ' . $this->getUsername());
+		}
+
+		return null;
+	}
+
+	/**
+	 * @return string[]
+	 */
+	public function getMemberOfGroups() {
+		if (isset($this->ldapEntry['memberof'])) {
+			return $this->ldapEntry['memberof'];
+		}
+		return [];
```

Comment:
I'd rather return something different because empty result is also a valid result, it would be difficult to distinguish this case and an error case (or if there isn't such entry)